src/main/java/io/nats/client/impl/NatsConnectionReader.java

```diff
@@ -98,17 +98,24 @@ void start(Future<DataPort> dataPortFuture) {
     // Returns a future that is completed when the thread completes, not when this
     // method does.
     Future<Boolean> stop() {
-        this.running.set(false);
-        if (dataPort != null) {
-            try {
-                dataPort.shutdownInput();
-            } catch (IOException e) {
-                // we don't care, we are shutting down anyway
+        if (running.get()) {

```

Comment:
If it's not running, this has already been called, no need to go into the logic.